### ISSUE TITLE
Custom error messages sometimes wont appear and some fields will not get marshalled

### ISSUE THREAD

**Issue by cachesdev** [Oudwins/zog#23]
been playing around with the library, and I have some bizarre errors I can't figure out. test code (echo handler):
```go
func (ah *apiHandlers) CreateUser(logger zerolog.Logger, repos *repositories.Repos) echo.HandlerFunc {
	return func(c echo.Context) error {
		input := make(map[string]any)

		err := c.Bind(&input)
		if err != nil {
			logger.Error().Err(err).Msg("cuerpo de request invalido")
			return echo.NewHTTPError(http.StatusBadRequest, "cuerpo de request invalido")
		}
                // This always prints correctly, the issue is not echo
		logger.Debug().Any("input", input).Msg("Echo parsed input map")

		var output struct {
			Nombre   string `zog:"nombre"`
			Apellido string `zog:"apellido"`
			Email    string `zog:"email"`
			AluID    int    `zog:"alu_id"`
			Password string `zog:"password"`
		}

		schema := z.Struct(z.Schema{
			"nombre":   z.String().Required(z.Message("this doesn't display even if validation fails")),
			"apellido": z.String().Required(),
			"email":    z.String().Required(),
			"alu_id":   z.Int().Required(), // this field specifically, is always empty
			"password": z.String().Required(),
		})

		errMap := schema.Parse(input, &output)
		if errMap != nil {
			return c.JSON(http.StatusBadRequest, z.Errors.SanitizeMap(errMap))
		}

		return c.JSON(200, output)
	}
}
```

if a post request with the following data is sent, the client receives mostly correct data back with exception of `AluID`:
```json
// Post request body
{
  "nombre": "Juan",
  "apellido": "Perez",
  "email": "test@test.com",
  "alu_id": 25,
  "password": "hunter1"
}

// Response
{
  "Nombre": "Juan",
  "Apellido": "Perez",
  "Email": "test@test.com",
  "AluID": 0,
  "Password": "hunter1"
}
```

the other issue is that when I try to change the err message with `z.Message()` for a `.String().Required()` validation, I will get the default message `is required` instead of my custom message:
```go
"nombre":   z.String().Required(z.Message("this doesn't display even if validation fails")),
```
Response when given an empty string:
```json
// Request body
{
  "nombre": "",
  "apellido": "Perez",
  "email": "test@test.com",
  "alu_id": 25,
  "password": "hunter1"
}

// Response
{
  "$first": [
    "is required"
  ],
  "nombre": [
    "is required"
  ]
}
```

i'm not sure if I made a mistake or if this is a library error

**Comment by Oudwins** [comment]
Hey!

Thanks for submitting the issue. Having had like 15 mins to look at these. These look like bugs on my end. I'll try to get them fixed today.

**Comment by Oudwins** [comment]
Up on further looking into it, The z.Message() is an issue. The AluId is actually you misunderstanding what the schema represents. However, Zog should probably panic in this case so I'll make a PR to make that panic.

Basically you are telling Zog to expect a struct field called `Alu_id`, what you want to do is either `aluID` or `AluID` in the schema. The zog tag is what handles the name in the json. Not sure if I am making sense?

The schema name should be the same (with optional lowercase first letter) as your struct field

**Comment by Oudwins** [comment]
I could probably write some better docs on the schema names though

**Comment by cachesdev** [comment]
> Up on further looking into it, The z.Message() is an issue (already fixed, but haven't pushed the release yet). The AluId is actually you misunderstanding what the schema represents. However, Zog should probably panic in this case so I'll make a PR to make that panic.
> 
> Basically you are telling Zog to expect a struct field called `Alu_id`, what you want to do is either `aluID` or `AluID` in the schema. The zog tag is what handles the name in the json. Not sure if I am making sense?
> 
> The schema name should be the same (with optional lowercase first letter) as your struct field

I see, I tested and now it does work in my mind I had to name the schema fields the same as the json 😅
but it seems to only work with lowercase struct field names, I am not sure if that's a bug or another misunderstanding
```go
// For this type
type User struct {
	Nombre   string   `json:"nombre" zog:"nombre"`
	Apellido string   `json:"apellido" zog:"apellido"`
	Email    string   `json:"email" zog:"email"`
	AluID    int      `json:"alu_id" zog:"alu_id"`
}

// this schema will return all empty fields in the response
		errMap := z.Struct(z.Schema{
			"Nombre":   z.String().Required(),
			"Apellido": z.String().Required(),
			"Email":    z.String().Required(),
			"AluID":    z.Int().Required(),
		}).Parse(input, &user)

// this schema works, returns back the same json as inputted.
		errMap := z.Struct(z.Schema{
			"nombre":   z.String().Required(),
			"apellido": z.String().Required(),
			"email":    z.String().Required(),
			"aluID":    z.Int().Required(),
		}).Parse(input, &user)
```

```json
// uppercase schema fields (exact match to struct)
{
  "nombre": "",
  "apellido": "",
  "email": "",
  "alu_id": 0
}

// lowecase schema fields
{
  "nombre": "Juan",
  "apellido": "Perez",
  "email": "test@teest.com",
  "alu_id": 25
}
```

**Comment by Oudwins** [comment]
Hey. Sorry I forgot to mention that uppercase first letter is actually something that I implemented for the next release. So its not yet in Zog. I'll do the release tonight and you can test it if you like but it is working correctly on my end.

On a completely unrelated topic. I want to create a package that will contain the error translations for spanish at some point. If you want to participate or share feedback that would be awesome

**Comment by cachesdev** [comment]
> Hey. Sorry I forgot to mention that uppercase first letter is actually something that I implemented for the next release. So its not yet in Zog. I'll do the release tonight and you can test it if you like but it is working correctly on my end.
> 
> On a completely unrelated topic. I want to create a package that will contain the error translations for spanish at some point. If you want to participate or share feedback that would be awesome

actually a package like that would reduce a lot of boilerplate when returning errors, I like the idea a lot. I am willing to do some spellchecking and such if needed, I am not too familiar yet with the library to contribute to it (i've cloned it and was reading the code yesterday)

you could have a similar system to how sql drivers are registered, and in each schema there could be an option to pick a language. so for example if I register the package for english and spanish, I could just pull the language from the request context and pass this variable to the zog option, and it will automatically return the correct language. would make changing the error language really easy for the library consumer but it will put a burden on the library maintainer (having to maintain all of the language packages)

opening a proposal issue may be a good idea to keep track of this properly

**Comment by Oudwins** [comment]
I plan on supporting both English and Spanish (soy de españa) any other language will have to go in a separate repo unless someone agrees to maintain the translation. 

I want to do skme internal  refactorings first but afterwards it should be very simple to implement. If you would be open to translating the errors map found in the config package that would be a great help

**Comment by cachesdev** [comment]
yeah I can do that for sure, i'll be keeping track of the repo.  I am latino myself, will be good to have two perspectives

**Comment by Oudwins** [comment]
Nice! I'll ping you on #26 when I am ready to go forward with this.

By the way, I have just merged v0.8 which has the fixes related to this issue so I am going to close this. Let me know if you find anything else. See PR #19 for changes. No breaking changes for this release
